<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CcVideoUIClass</class>
 <widget class="QWidget" name="CcVideoUIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CcVideoUI</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#CcVideoUIClass
{
    background-color: #606060; /* 浅灰色背景 */
    border-radius: 15px; /* 圆角 */
    border: 2px solid black; /* 黑色边框 */
}

#Close_btn {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ff7f7f, stop: 1 #ff0000); /* 从上到下的渐变 */
    border-radius: 8px; /* 圆角 */
    color: white; /* 设置文字颜色为白色 */
    padding: 3px; /* 设置内边距 */
    font-size: 11px; /* 设置字体大小 */
    font-weight: bold; /* 设置字体加粗 */
}

#Close_btn:hover {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 1, y2: 1,
                                      stop: 0 #ff6666, stop: 1 #cc0000); /* 悬停时红色更深 */
}

#Close_btn:pressed {
    background-color: qlineargradient(x1: 0, y1: 0, x2: 1, y2: 1,
                                      stop: 0 #ff4d4d, stop: 1 #990000); /* 按下时红色更深 */

}

/* 默认状态：从上到下，白色到黑色的渐变色 */
QPushButton {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,  /* 垂直渐变方向 */
        stop:0 #a0a0a0, stop:0.4 #505050, stop:0.6 #505050, stop:1 #a0a0a0
    );
    border: 2px solid #a0a0a0;     /* 黑色边框 */
    border-radius: 5px;         /* 圆角 */
    color: white;                /* 文字颜色 */
    padding: 3px;               /* 内边距 */
    font-size: 11px;             /* 字体大小 */
    font-weight: bold;           /* 字体加粗 */
}

/* 鼠标悬停状态：从上到下，黑色到白色的渐变色（与默认状态取反） */
QPushButton:hover {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,  /* 垂直渐变方向 */
        stop:0 #505050, stop:0.4 #a0a0a0, stop:0.6 #a0a0a0, stop:1 #505050
    );
    color: white;                /* 文字颜色 */
}

/* 按下状态：从上到下，灰色到深灰色的渐变色 */
QPushButton:pressed {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,  /* 垂直渐变方向 */
        stop:0 #505050,          /* 顶部灰色 */
        stop:0.5 #808080,        /* 中间灰色 */
        stop:1 #505050           /* 底部深灰色 */
    );
    color: #a0a0a0;                /* 文字颜色 */
    padding: 11px 9px 9px 11px;  /* 模拟按下效果 */
    border: 2px solid #404040;   /* 边框颜色变深 */
}

QLabel{
	color: rgb(255,255,255);
	font: 75 12pt &quot;黑体&quot;;
}

#Play_btn{
	background-color: rgba(255,255,255,0);
}

#Play_btn:!hover{
	image: url(:/CcVideoUI/res/startNor.png);
	border: none;
}	

#Play_btn:hover{
	image: url(:/CcVideoUI/res/startpress.png);
	border: none;
}	

#Play_btn:pressed{
	image: url(:/CcVideoUI/res/startpress.png);
	border: none;
}	

#Stop_btn{
	background-color: rgba(255,255,255,0);
}

#Stop_btn:!hover{
	image: url(:/CcVideoUI/res/stopNor.png);
	border: none;
}	

#Stop_btn:hover{
	image: url(:/CcVideoUI/res/stoppress.png);
	border: none;
}	

#Stop_btn:pressed{
	image: url(:/CcVideoUI/res/stoppress.png);
	border: none;
}	</string>
  </property>
  <widget class="QPushButton" name="Close_btn">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>0</y>
     <width>21</width>
     <height>21</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string>×</string>
   </property>
  </widget>
  <widget class="CcVideoWidget" name="src1">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>380</width>
     <height>261</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="Open_btn">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>300</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>打 开</string>
   </property>
  </widget>
  <widget class="QSlider" name="playSlide">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>300</y>
     <width>381</width>
     <height>18</height>
    </rect>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Orientation::Horizontal</enum>
   </property>
  </widget>
  <widget class="QWidget" name="widget_2" native="true">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>350</y>
     <width>461</width>
     <height>241</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>8</number>
    </property>
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <layout class="QGridLayout" name="gridLayout">
       <property name="bottomMargin">
        <number>4</number>
       </property>
       <property name="verticalSpacing">
        <number>2</number>
       </property>
       <item row="7" column="3">
        <widget class="QSpinBox" name="pydown_box">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>图像镜像</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0" colspan="2">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>图像裁剪(X,Y,W,H)</string>
         </property>
        </widget>
       </item>
       <item row="9" column="4">
        <widget class="QDoubleSpinBox" name="mark_a">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>23</height>
          </size>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="3" column="2" colspan="2">
        <widget class="QComboBox" name="flip_box">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <item>
          <property name="text">
           <string>不处理</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>左右</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>上下</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>上下左右</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>亮度[0~100]</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="3">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>图像金字塔(高斯、拉普拉斯)</string>
         </property>
        </widget>
       </item>
       <item row="9" column="3">
        <widget class="QSpinBox" name="mark_y">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item row="0" column="4" colspan="2">
        <widget class="QPushButton" name="Set_btn">
         <property name="minimumSize">
          <size>
           <width>65</width>
           <height>30</height>
          </size>
         </property>
         <property name="text">
          <string>设置</string>
         </property>
        </widget>
       </item>
       <item row="7" column="4">
        <widget class="QSpinBox" name="pyup_box">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="9" column="0" colspan="2">
        <widget class="QPushButton" name="mark_btn">
         <property name="text">
          <string>水印(x,y,α)</string>
         </property>
        </widget>
       </item>
       <item row="8" column="4">
        <widget class="QSpinBox" name="clip_w">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item row="2" column="2" colspan="2">
        <widget class="QComboBox" name="rotate_box">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <item>
          <property name="text">
           <string>0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>90</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>180</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>270</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="5" column="0" colspan="3">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>图像尺寸(宽、高)</string>
         </property>
        </widget>
       </item>
       <item row="8" column="2">
        <widget class="QSpinBox" name="clip_x">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item row="0" column="2" colspan="2">
        <widget class="QSpinBox" name="bright">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="8" column="3">
        <widget class="QSpinBox" name="clip_y">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>图像旋转</string>
         </property>
        </widget>
       </item>
       <item row="9" column="2">
        <widget class="QSpinBox" name="mark_x">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item row="5" column="4">
        <widget class="QSpinBox" name="height_box">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item row="8" column="5">
        <widget class="QSpinBox" name="clip_h">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2" colspan="2">
        <widget class="QDoubleSpinBox" name="contrast">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>23</height>
          </size>
         </property>
         <property name="minimum">
          <double>1.000000000000000</double>
         </property>
         <property name="maximum">
          <double>3.000000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="5" column="3">
        <widget class="QSpinBox" name="width_box">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>对比度[1.0~3.0]</string>
         </property>
        </widget>
       </item>
       <item row="10" column="2">
        <widget class="QDoubleSpinBox" name="blend_box">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>23</height>
          </size>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="value">
          <double>0.500000000000000</double>
         </property>
        </widget>
       </item>
       <item row="10" column="0" colspan="2">
        <widget class="QPushButton" name="blend_btn">
         <property name="text">
          <string>融合视频</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="CcVideoWidget" name="dst">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>20</y>
     <width>380</width>
     <height>261</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="Export_btn">
   <property name="geometry">
    <rect>
     <x>490</x>
     <y>300</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>导出</string>
   </property>
  </widget>
  <widget class="QPushButton" name="Play_btn">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>330</y>
     <width>71</width>
     <height>51</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="Stop_btn">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>330</y>
     <width>91</width>
     <height>51</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QComboBox" name="color_btn">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>300</y>
     <width>81</width>
     <height>25</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>25</height>
    </size>
   </property>
   <item>
    <property name="text">
     <string>RGB图像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>GRAY图像</string>
    </property>
   </item>
  </widget>
  <widget class="CcVideoWidget" name="src2">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>380</y>
     <width>320</width>
     <height>220</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="merge_btn">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>340</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>合并视频</string>
   </property>
  </widget>
  <widget class="QSlider" name="left_silde">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>280</y>
     <width>381</width>
     <height>18</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QSlider::handle:horizontal{
	  background: #b0b0b0;
      border-radius: 4px;
}</string>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Orientation::Horizontal</enum>
   </property>
  </widget>
  <widget class="QSlider" name="right_slide">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>320</y>
     <width>381</width>
     <height>18</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QSlider::handle:horizontal{
	  background: #b0b0b0;
      border-radius: 4px;
}</string>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="value">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Orientation::Horizontal</enum>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CcVideoWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>ccvideowidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="CcVideoUI.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>Close_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>785</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Open_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>Open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>325</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlide</sender>
   <signal>sliderPressed()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>SliderPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>315</x>
     <y>311</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlide</sender>
   <signal>sliderReleased()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>SliderRelease()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>241</x>
     <y>312</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlide</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>SetPos(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Set_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>Set()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>779</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>622</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Export_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>Export()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>490</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>-19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Play_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>Play()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>213</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>215</x>
     <y>-38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Stop_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>Pause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>340</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>-15</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mark_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>Mark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>189</x>
     <y>-16</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blend_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>Blend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>369</x>
     <y>575</y>
    </hint>
    <hint type="destinationlabel">
     <x>254</x>
     <y>-23</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>merge_btn</sender>
   <signal>clicked()</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>Merge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>572</x>
     <y>-100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>left_silde</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>LeftSide(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>-49</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>right_slide</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>CcVideoUIClass</receiver>
   <slot>RightSide(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>152</x>
     <y>327</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>-71</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Open()</slot>
  <slot>SliderPress()</slot>
  <slot>SliderRelease()</slot>
  <slot>SetPos(int)</slot>
  <slot>Set()</slot>
  <slot>Export()</slot>
  <slot>Play()</slot>
  <slot>Pause()</slot>
  <slot>Mark()</slot>
  <slot>Blend()</slot>
  <slot>Merge()</slot>
  <slot>LeftSide(int)</slot>
  <slot>RightSide(int)</slot>
 </slots>
</ui>
